hal/src/sleeping_delay.rs

```diff
@@ -35,6 +39,34 @@ where
     }
 }
 
+#[cfg(feature = "embedded-hal-1")]
+impl<TIM> DelayNs for SleepingDelay<TIM>
+where
+    TIM: InterruptDrivenTimer,
+{
+    fn delay_ns(&mut self, ns: u32) {
+        // Determine how many cycles we need to run for this delay, if any
+        // Avoid timers that run longer than a second because for 48 MHz-based timers,
+        //   there is no valid divisor + cycle count greater than ~1.3s, so we'd panic.
+        let mut loop_count: u32 = 1 + (ns / NUM_NS_IN_S);
```

Comment:
I think our timers take a u32 nanosecond count, so this loop may be able to be removed. Need to verify